pkg/virtual/framework/forwardingregistry/wrappers.go

```diff
@@ -78,3 +79,37 @@ func WithLabelSelector(labelSelectorFrom func(ctx context.Context) labels.Requir
 		return storage
 	}
 }
+
+func WithStaticLabelSelectorAndInWriteCallsCheck(labelSelector labels.Requirements) StorageWrapper {
+	return func(resource schema.GroupResource, storage *StoreFuncs) *StoreFuncs {
+
+		delegateCreater := storage.CreaterFunc
+		storage.CreaterFunc = func(ctx context.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
+			if meta, ok := obj.(metav1.Object); ok {
+				if !labels.Everything().Add(labelSelector...).Matches(labels.Set(meta.GetLabels())) {
+					return nil, errors.NewBadRequest(fmt.Sprintf("label selector %q does not match labels %v", labelSelector, meta.GetLabels()))
+				}
+			}
+			return delegateCreater.Create(ctx, obj, createValidation, options)
+		}
+
+		delegateUpdater := storage.UpdaterFunc
+		storage.UpdaterFunc = func(ctx context.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc, forceAllowCreate bool, options *metav1.UpdateOptions) (runtime.Object, bool, error) {
+			obj, _, err := delegateUpdater.Update(ctx, name, objInfo, createValidation, updateValidation, forceAllowCreate, options)
+			if err != nil {
+				return obj, false, err
+			}
+
+			if meta, ok := obj.(metav1.Object); ok {
```

Comment:
Not sure it really makes sense to do the check after the update on the delegate here.
However after thinking a bit more to it, the implementation of the `Update` in the ForwardingRegistry  does a `Get` in any case. So if the updated object would not satisfy the label selector, the Get wold already return a `NotFound` error afaict.
So it's probably sufficient to only restrict on `Create`.

Reply:
Well, that's not entrirely true.
The check should be done, but :
- before the update on delegate
- on non-nil object returned by `objInfo.UpdatedObject(ctx, nil)`

This way the check will still be done in case of:
- Update request with "AllowCreate" option
- Server-side apply request that will trigger creation.

Reply:
sorry about that, updated and fixed, also added a new test to cover the update case.